ESP8266 Master  read/write Hreg

Barone1968 opened the pull request:
I tried to use your library to configure ESP8266 master, (example Coil only) and certainly I have not been able, can you help me, I tried to Use hreg but nothing..thanks

emelianov committed:
Stability fixes

emelianov committed:
Switch to std::vector

emelianov committed:
Documentation changes

emelianov committed:
ModbusIP header cleanup

emelianov committed:
Set zerro base for remote registers addressing

emelianov committed:
adsReg fix

emelianov committed:
Documentation fixes

emelianov committed:
Documentation changes

emelianov committed:
Modbus Master implementation

emelianov committed:
Mistype correction

emelianov committed:
Mistype correction

emelianov committed:
Update README.md

emelianov committed:
Wrong bit-array processing fix (get/setMultipleBits)

emelianov committed:
Merge branch 'master' of https://github.com/emelianov/modbus-esp8266

emelianov committed:
Global regs definition

emelianov committed:
Comments and address expansion core types

emelianov committed:
Add TAddress +/++/+= overloads

emelianov committed:
Expand register addressing to 0..65535

emelianov committed:
Experimental: ModbusSerial (slave)

emelianov committed:
Examples: Callback removed changes made by exident

emelianov committed:
Experimental: ModbusSerial (slave)

emelianov committed:
Add destructors for Modbus and TTransaction

emelianov committed:
Bugfix and API change and expansion

Fixed crash at transaction timeout
Change transaction callback function arguments
Prepare to add readCoil/Ists/Ireg/Hreg
Add writeHreg for multiple registers write
pull/push/read/write* methods returns transaction id or 0 on error
Implement removeCoil/etc methods
remove lastTransaction method

emelianov committed:
Remove ModbusSerial.*

emelianov committed:
Implement remote slave read/wirte for all register's types

emelianov committed:
Remove ModbusSerial.*

emelianov committed:
Add multiple registers fix, API change (pull/push regs)

emelianov committed:
ESP32 compile error fix

emelianov committed:
push\pull code cleanup after last API changes

emelianov committed:
Implement pullHregToIreg, pullCoilToIsts, pushIstsToCoil, pushIregToHreg

emelianov committed:
Optimize std::vector processing

emelianov committed:
Extend removeCoil/Hreg/... to remove multiple registers

emelianov committed:
Add methods

emelianov committed:
Quick fix for new API

emelianov committed:
Callback API extended to support multiple handlers for event

PeterEmbedded committed:
Update Master.ino

Fixed some typos

PeterEmbedded committed:
Update AnalogInput.ino

Fixed a typo

emelianov committed:
Merge pull request #6 from PeterEmbedded/patch-2

Update AnalogInput.ino

emelianov committed:
Merge pull request #5 from PeterEmbedded/patch-1

Update Master.ino

emelianov committed:
Mistype fix

emelianov committed:
Merge branch 'master' of https://github.com/emelianov/modbus-esp8266

emelianov committed:
Finishing with multiple callbacks on event

emelianov committed:
Add HReg read example

emelianov committed:
Wrong byte order in read/writeHreg, read/writeIreg fixed

emelianov committed:
Static client's array

emelianov committed:
add Modbus Unit parameter to master calls

emelianov committed:
Add Modbus unit parameter to read/write/pull/push. Remove responce wait if no CallBack

emelianov committed:
Add autoConnect for read/write/pull/push

emelianov committed:
Add autoConnect for read/write/pull/push

Commit:
Merge remote-tracking branch 'origin/alternate'

emelianov committed:
2.0

emelianov committed:
Change comments

emelianov committed:
Docs changes

emelianov committed:
Fix examples

emelianov committed:
Fix crash on disconnect (with 2.5.x)

emelianov committed:
Fix readCoil/Hreg/Ists/Ireg

emelianov committed:
2.0.1

emelianov committed:
Fix master write multile Hregs/Coils. Change TCP flushing

emelianov committed:
Merge branch 'master' of https://github.com/emelianov/modbus-esp8266

emelianov committed:
Disconnect initial implementation

emelianov committed:
Merge branch 'master' of https://github.com/emelianov/modbus-esp8266

emelianov committed:
2.1 BETA

emelianov committed:
dropTransaction()/clean() refactoring

emelianov committed:
Fix writeMiltipleBits for array

emelianov committed:
Update examples

emelianov committed:
Limit incoming connections to one per IP

emelianov committed:
Switch to leonardo/mega

emelianov committed:
Comments added

emelianov committed:
Remove useless ->stop() lines

emelianov committed:
ModbusRTU slave initial draft

emelianov committed:
Update doc

emelianov committed:
ModbusRTU task() refactoring

emelianov committed:
Merge slave and master to ModbusRTU

emelianov committed:
Move CRC table to PROGMEM

emelianov committed:
Move crc routine inside ModbusRTU class, test Master methods added, task() fix for Master

emelianov committed:
Master readCoil tested

emelianov committed:
Switch to templates, add HardwareSerial support

emelianov committed:
Add rest of master methods

emelianov committed:
Add ModbusRTU examples

emelianov committed:
Fix mistype

emelianov committed:
Extract write to rawSend() routine. Use internal TX enable control for SoftwareSerial

emelianov committed:
Fix count of processing regs/coils/etc according to Modbus Spacification

emelianov committed:
Fix read/pullCoils/Ists number limitation. Cleanup.

emelianov committed:
Merge branch 'master' of https://github.com/emelianov/modbus-esp8266

emelianov committed:
Fix rawSend()

emelianov committed:
Revert to no-template

emelianov committed:
ReadMe update

emelianov committed:
Update RTU examples

emelianov committed:
Remove SoftwareSerial for ESP32

emelianov committed:
API documentation update

emelianov committed:
Update README.md

emelianov committed:
Update README.md

emelianov committed:
Update README.md

emelianov committed:
Allow use just begin(&Serial1) on already initialized port

emelianov committed:
Merge branch 'master' of https://github.com/emelianov/modbus-esp8266

emelianov committed:
Update serial init in examples

emelianov committed:
Generalize eventSource() and implement it for RTU.

emelianov committed:
Generalize eventSource() and implement it for RTU.

emelianov committed:
sendRaw(): Move delyay() after txPin LOW

emelianov committed:
Change .begin() API call

emelianov committed:
Notes added

per1234 committed:
Use correct field separator in keywords.txt

The Arduino IDE requires the use of a single true tab separator between the keyword name and identifier. When spaces are used rather than a true tab, the keyword is not highlighted.

Reference:
https://github.com/arduino/Arduino/wiki/Arduino-IDE-1.5:-Library-specification#keywords

emelianov committed:
Merge pull request #16 from per1234/keywords-separator

Use correct field separator in keywords.txt

emelianov committed:
Text update

emelianov committed:
Merge branch 'master' of https://github.com/emelianov/modbus-esp8266

emelianov committed:
Replace magin number for buffer size

emelianov committed:
Allow to change TCP port for master/slave

emelianov committed:
Include byteswap.h only for ESP32

emelianov committed:
Fix cleanup() crash. Thanks to Darius. Fix task() crash on wrong packet.

emelianov committed:
Merge branch 'master' of https://github.com/emelianov/modbus-esp8266

emelianov committed:
task() error processing fixes

emelianov committed:
task() cleanup

emelianov committed:
Misc

emelianov committed:
task(): Dramaticly increase data exchange rate

emelianov committed:
ESP32-related build fixes

emelianov committed:
ModbusIP task(): Limit execution time & fix possible deadlock

emelianov committed:
Fix write multiple Hregs. Refactor internal *MultipleWords()

emelianov committed:
Fix crash on no callback function on read\write remote. Thanks to Krutarth Trivedi

emelianov committed:
Write Coil(s)/Hreg(s) tests added

emelianov committed:
Read Coils/Hregs/Iregs/Istss tests added

emelianov committed:
3.0.0

emelianov committed:
ESP32: Add critical section to escape possible timouts

emelianov committed:
TAddress != operator added

brainelectronics committed:
Support non ESP devices (#40)

* may handover baudrate to begin serial communication, use baudRate() only on ESP32 or ESP8266

* include ModbusIP_ESP8266 only if board is ESP32 or ESP8266

* setBaudrate function added, fixed PR remarks

* fixed begin function in header

* small fix for setBaudrate function

* updated RTU examples for ESP32, ESP8266, STM32F103

emelianov committed:
ModburRTU: setBaudrate() rewrite

emelianov committed:
Add incoming packet output to serial for debug

emelianov committed:
README update

emelianov committed:
README update

emelianov committed:
Remove code duplication in masterPDU

emelianov committed:
ModbusIP: master/slave => client/server

emelianov committed:
ReadMe update

emelianov committed:
ModbusRTU: ESP32 possible recive fix

emelianov committed:
ESP32: Drop mutex in task as it were resulting stability problems

emelianov committed:
ModbusRTU task(): reduce timeout possibility due to rare task() calls

emelianov committed:
ModbusIP task(): reduce timeout possibility due to rare task() calls

emelianov committed:
Put back global registers removed by mistake

emelianov committed:
3.0.1

emelianov committed:
Update README.md

emelianov committed:
ModbusTCP Client: ESP32 fix unexpected transaction timeout

emelianov committed:
Merge branch '3.0.1'

emelianov committed:
Merge branch 'master' of https://github.com/emelianov/modbus-esp8266

emelianov committed:
4.0.0.DEVEL
- Based on 3.0.2
- Merge API implementation for RTU and TCP
- Split TCP implementation from specific TCP Stack libraryes
- Templates for TLS and W5x00 support
- New flexible API preparations

emelianov committed:
Add Modbus read/write file block support
- 0x14 + test
- 0x15 + test
- 0x17 (Mask register)

emelianov committed:
W5x00 Ethernet support (not tested)
- Added missing 3.0.2 changes
- ModbusTCPTemplate changes to support Ethernet 1.x

emelianov committed:
no-STL build support (not completly tested)
- Add basic class to replace std::vector
- Changed way to W5x0 Ethernet library support

emelianov committed:
Modbus/TCP Security implemantation and API funcs
- Client/Server for ESP8266
- Client for ESP32 (not tested)
- TLS exmample (including demo cerificates)
- API extended to allow connect to TCP/TLS server by name (for ESP and
Ethernet both)
- non-STL mode appeas to be broken

emelianov committed:
ModbusRTU: Add inverse logic support for transmit control

emelianov committed:
Add request callback
- Callback prior operation. Can be used for acess control.
- Callback at operation successfuly finished. (Optional API)
- Example of using this kind of callback
- ModbusTCP with Ethernet library example
- Documentation and examples refactoring started

emelianov committed:
ModbusRTU class slight refactoring
- Add RX/TX pin inverse (form 3.0.3)
- Extended ESP32 critical section over .flush() to escape potential data
loss (from 3.0.4 ?)
- Generalized ModbusRTUTemplate class to allow SoftwareSerial under
ESP32

emelianov committed:
Cleanup code style
- Remove compile-time warning in mastedPDU()
- #include code style change
- Remove compile-time warning in structs

emelianov committed:
Cleanup code style and fix bugs
- Remove compile-time warning in mastedPDU()
- #include code style change
- Remove compile-time warning in structs
- Generic Arduino support finally added
- Examples and documentation refactoring (in progress)

emelianov committed:
W5x00 @ESP8266
- Split v1 and v2 derivated libraries classes
- Update Ethernet server example

emelianov committed:
API extension
- Set multiple local registers value from array
- Add multiple registers and fill values from array

emelianov committed:
Fixed compatibility with Ethernet library
- Usage of accept() instead available() for Ethernet
- Added wrapper accept() => available() for ESP
- Removed possible deadlock in task()
- Documentation and examples spet-by-step changes

emelianov committed:
ModbusRTU stability improvment
- .task() refactoring
- Fixed possible wrong transactional callback call
- Examples rearranged
- Documentation update (in-progress)

emelianov committed:
Critical fixes
- ModbusIP: Fixed missed server connection free in destructor
- Fixed addition of missing registers in .poll*() calls
- ModbusRTU: Fixed callback remains assigned after transaction ends

emelianov committed:
Fixes
- ModbusRTU: Fix transaction callback remains assigned after request end
- ModbusTCP: Free server connection in destructor
- Declare global registers and callbacks as ststic members
- ModbusRTU: Fixed possible packet lost because of wrong cleanup after
send
- ModbusRTU: Slight optimezation .task() for Master

emelianov committed:
ModbusRTU: Fix possible data loss and slight .task() optimezation
It were no cleanup of _len and _frame if .rawSend() is called with no
callback

emelianov committed:
Fix bridge example

emelianov committed:
Fix bridge example

emelianov committed:
ModbusRTU: Fix packet loss after sending attempt on busy bus

emelianov committed:
ModbusRTU: Fix packet loss after sending attempt on busy bus

emelianov committed:
ModbusTCP: Free buffers after request have sent

emelianov committed:
Rewrite all callbacks to be std::function

emelianov committed:
3.0.4

emelianov committed:
Fixed wrong onRequest/onRequestSuccess location

emelianov committed:
ModbusRTU Stability fixes
- Fixed early bus release if no transactional callback assigned for read/write Hreg/Coil
- Fixed possible responce loss for ESP32 master

emelianov committed:
Server: Fix callback called twice on register read

emelianov committed:
Port changes from 3.0.6
- ModbusRTU: Fix easly bus release if no callback
- Server: Fix onGet callback called twice
- Documentation light update

emelianov committed:
W5x00: Support for ESP32, Client example

emelianov committed:
4.0.0 Development release

emelianov committed:
Create README.md

emelianov committed:
Fix __bswap_16 rename

emelianov committed:
ESP32: Correct task switch blocking.
- Add multitask example

emelianov committed:
ESP32 RTU-Concurent Example: Change Binary to Mutex

emelianov committed:
ModbusTCP Client: Fix persistent connection error in some situations
ModbusRTU Client: Add sync read example

emelianov committed:
Full Bridge helping changes (stage #1)

emelianov committed:
Fix add register on pull. Fix TLS.

emelianov committed:
Move Adruino Due to STL

emelianov committed:
Merge branch 'master' of github.com:emelianov/modbus-esp8266

emelianov committed:
Temprary disable Arduino Due STL due to PlatformIO incompatibility

emelianov commented:
> I tried to use your library to configure ESP8266 master, (example Coil only) and certainly I have not been able, can you help me, I tried to Use hreg but nothing..thanks

@Barone1968 
Please create issue under https://github.com/emelianov/modbus-esp8266/issues and provide more details on the problem.

emelianov committed:
Fix compile error issue #112

emelianov committed:
Add EX_SLAVE_FAILURE on multiplr registers write error

emelianov committed:
Change on request callback API

TienHuyIoT committed:
Fix Esp32 crash and master broadcast ID failure

Details:
    - _slaveId must be difference 0 value to cleanup function run and return true.
    - Not using vTaskSuspendAll() and xTaskResumeAll() by redefine ESP32
    to ESP32_REMOVE.
    - Update example ESP32-Concurent.ino and using Tool Modbus Slave on
    PC for test.

Refs:
    - Tool Modbus Slave link: https://www.modbustools.com/download.html
    - ESP32 Arduino Release 1.0.6 based on ESP-IDF v3.3.5
      https://github.com/espressif/arduino-esp32/releases/tag/1.0.6

emelianov committed:
Fix memory leak on broadcast with transactional callback

emelianov committed:
Fix ESP32 RTU
- Remove vSuspendTask commited by excident
- Fix potential crash in low memory conditions
- Misc examples changes

emelianov committed:
Merge branch 'master' of github.com:emelianov/modbus-esp8266

emelianov committed:
FC_READWRITE_REGS implementation

emelianov committed:
Slave core refactoring
- "Magic" returns from internal readWords/readBits eliminated
- Check out of range on incoming requests
- Fix possible out of range effects on adding registers

escherstair committed:
Fix links in README.md (#137)

emelianov committed:
Delete ModbusRTU.cpp

emelianov committed:
Documentation changes

emelianov committed:
Revert ModbusRTU.cpp

TienHuyIoT committed:
ESP32-Concurent.ino Modifications (#132)

Details:
    - Update example ESP32-Concurent.ino and using Tool Modbus Slave on
    PC for test.

emelianov committed:
Documentation update

emelianov committed:
Update README.md

emelianov committed:
Prapare for release
- Test and Fix ReadWrite Hregs
- Test and Fix Mask Hreg
- Add Firmware Update example
- Fix onRequest arguments
- Update documentation

emelianov committed:
Documentation fix

emerout committed:
Fix broken link and typo in example/RTU/Readme (#138)

Co-authored-by: Emeric Chardiny <emeric@ecy-conseil.com>

emelianov committed:
Create FUNDING.yml

emelianov committed:
Finalize FW-Update example

emelianov committed:
ESP32: Removed taskENTER_CRITICAL and vTaskSuspendAll()

emelianov committed:
ESP32: Removed taskENTER_CRITICAL and vTaskSuspendAll()

vondraussen committed:
fix warning (#161)

* fix warning
comparison of int and unsigned int
* convert tabs to spaces
to remove warnings -Wmisleading-indentation

emelianov committed:
4.0.0

emelianov committed:
Fix pushCoil and pushIstsToCoil (#162)

gonzabrusco committed:
Millis to Micros in RTU (#170)

* Millis to Micros in RTU

* Replace setBaudrate with calculateMinimumInterFrameDelay

* Change the word Delay for Time

* Changes suggested applied

Co-authored-by: Gonzalo Brusco <gonzalo@powermeter.com.ar>

emelianov committed:
Fix example mistype

emelianov committed:
Merge branch 'master' of github.com:emelianov/modbus-esp8266

emelianov committed:
4.1.0.DEV (#174)

* ModbusRTU: Separate pins for RE/DE support

* ModbusRTU: RE/DE receive fix

* Examples: Added ClearCode ModbusTCP server/client

* Fix CleadCore example client/server was swapped

* ClearCore example: remove unneeded lines

* ClearCore example: fix

* MODBUSIP_DEBUG added

* Fix example mistype

emelianov committed:
Raw callback (#175)

* True bridge core implementation test

* True bridge example

* Make works: Raw callbacks and True bridge example.

* Add debuglog output to True Bridge example

* Fix mistipe

* Refactor raw callback flow in task()

* Update True Bridge example

* Set MODBUSIP_CONNECT_TIMEOUT optional

* Fix example mistype

emelianov committed:
Merge branch 'master' of github.com:emelianov/modbus-esp8266

emelianov committed:
Fix merge defects

emelianov committed:
'-' => '_' in filenames

emelianov committed:
Minor changes:
ModbusTCP: Make DNS as Modbus target optional
ModbusRTU: Add extra deleay for RD DE control
API-RawCallback: Add force packet processing

emelianov committed:
Add Multipe ID response example

emelianov committed:
_ to - in examples

emelianov committed:
4.1.0-RC1

mako777 committed:
Correct indentation (#190)

Compilation warning:
  265 |      continue;

emelianov committed:
ModbusTCP: Fix Raw callback processing

emelianov committed:
ModbusTCP to ModbusRTU example added

emelianov committed:
4.1.0 release
- Fix TransactionId
- Misc documentation fixes

emelianov committed:
Memory leak, examples and API fixes
- Fix memory leak (issue #208)
- Extend cbEnable/cbDisable API
- Fix TCP-to-RTU example

emelianov committed:
Replace magic number with constants
- MODBUS_MAX_WORDS/BITS
- NULLREG
- Alternative API read/write functions implemented

alepiva committed:
Update ESP32-Concurent.ino (#217)

typo fix

Commit:
Fix typo (#218)

Commit:
Fix typo (#219)

emelianov committed:
ModbusEthernet: Portenta and ESP32 boards build fix

emelianov committed:
TCP-to-RTU example: Fix cbTcpRaw

emelianov committed:
TCP-to-RTU example: Fix timeout processing

alexbarcelo committed:
Minor grammar / typos (#249)

emelianov committed:
Respond EX_ILLEGAL_ADDRESS on non-existent reg

emelianov committed:
README update

rob040 committed:
Fix deadlinks in example docs #272 (#275)

emelianov committed:
Fix Slave answer with malformed packet #268

emelianov committed:
STM32 Ethernet and simular workaround #278

emelianov committed:
txPin to txEnablePin, direct to txEnableDirect #266

emelianov committed:
Uncomment the explicit stop() method call to allow sockets cleanups (#281)

This should allow w5xxx-based ethernet to clean up their sockets' states
upon disconnection, preventing the server implementation to experience
sockets starvation

Co-authored-by: Marco Fagiolini <mfx@amdx.de>

ArekKubacki committed:
Fixed error: modbus-esp8266/src/Modbus.cpp:318:25: error: comparison is always false due to limited range of data type [-Werror=type-limits] (#292)

emelianov committed:
RTU: Add _onRequest on illegal function

emelianov committed:
Merge branch 'master' of github.com:emelianov/modbus-esp8266

basisbit committed:
chore: correct comment in modbus client example (#342)

the esp was used as modbus client, but the code comment said modbus server.

alepiva committed:
correction of TCP debug (#326)

used MODBUSRTU_DEBUG insted of MODBUSIP_DEBUG

lmartorella committed:
- Added const (#313)

# Conflicts:
#	src/ModbusAPI.h

lmartorella committed:
- Allow to change MODBUSRTU_TIMEOUT from project (#315)

emelianov committed:
Merge branch 'master' of github.com:emelianov/modbus-esp8266

emelianov committed:
onGetRemove/onSetRemove fix

emelianov committed:
begin with re/de warning fix

smilexs4 committed:
Update TCP-to-RTU-Simulator.ino (#361)

Fixed typo
